### DISCUSSION TITLE
Lookign for ACIS and SAT format documenation

### DISCUSSION THREAD

**Discussion by NikhilSP** [mozman/ezdxf#1074] [General]
Hey, I am trying to load geometries in SAT file at my work. I need some help becuase I am not able to find anything useful online except pdf attached

Firstly, the text in dxf file was encrypted in two different formats .

Format 1 :
```
1
koo mn n o 
  1
ll P/:1 [:,681 ^ 336>1<: ^ \VL ]*63;:- nm ^ \VL kqoo QK mk R01 Y:= nm noelgejl momk 
  1
mjqlffffffffffffff fqfffffffffffffffj:rooh n:rono 
  1
=0;& {rn {n {rn {rn |

```

I decrypted the above code using 

```
private static char Decode(char c)
    {
        if (c == 32)
        {
            return c;
        }

        return (char)(159 - c);
    }
```

Format 2 : 

```
310
41534D2042696E61727946696C65341C57000000000000020000000400000007104175746F6
```

I haven't yet decrypted it successfully (tried Hex to text) but I saw your answer on stack overflow that they are using XOR encryption. I am going to try that today. 

After decryption, I get following code. 
```
body $-1 $1 $-1 $-1 #
 lump $-1 $-1 $2 $0 #
shell $-1 $-1 $-1 $3 $-1 $1 #
```

My problem is I am not able  to understand how this data format works and not able to find useful info in documentation and books.
Can you please help me here?

[sat.pdf](https://github.com/mozman/ezdxf/files/15190085/sat.pdf)

**Comment by mozman** [comment]
If you use the SAT format professionally, there is a much easier way. Your company should license the SDK from Spatial Inc. and you will get all the information and tools you need.

If this is not an option for you, you can search the sub-package `ezdxf.acis` for information. I don't have any more and as you have already noticed, there is no other sufficient publicly accessible documentation.
